composer/utils/object_store/oci_object_store.py

```diff
@@ -130,19 +144,45 @@ def download_object(
         dirname = os.path.dirname(filename)
         if dirname:
             os.makedirs(dirname, exist_ok=True)
-        tmp_path = str(filename) + f'.{uuid.uuid4()}.tmp'
-
-        try:
-            response = self.client.get_object(
-                namespace_name=self.namespace,
-                bucket_name=self.bucket,
-                object_name=object_name,
-            )
-        except Exception as e:
-            _reraise_oci_errors(self.get_uri(object_name), e)
 
-        with open(tmp_path, 'wb') as f:
-            f.write(response.data.content)
+        # Get the size of the object
+        head_object_response = self.client.head_object(self.namespace, self.bucket, object_name)
+        object_size = head_object_response.headers['content-length']
+        # Calculate the part sizes
+        base_part_size = int(object_size) // num_parts
+        remainder = int(object_size) % num_parts
+        part_sizes = [base_part_size] * num_parts
+        for i in range(remainder):
+            part_sizes[i] += 1
+        part_sizes = [part_size for part_size in part_sizes if part_size > 0]
+
+        tmp_part_dir = os.path.join(dirname, f'{str(filename)}-parts-{uuid.uuid4()}')
+        os.makedirs(tmp_part_dir, exist_ok=True)
+        with TemporaryDirectory(dir=tmp_part_dir) as temp_dir:
+            try:
+                # Download parts in parallel
+                parts = []
+                with concurrent.futures.ThreadPoolExecutor() as executor:
+                    futures = []
+                    start_byte = 0
+                    for i, part_size in enumerate(part_sizes):
+                        end_byte = start_byte + part_size - 1
+                        futures.append(
+                            executor.submit(self._download_part, object_name, temp_dir, start_byte, end_byte, i))
```

Comment:
Also not sure if this is what you want as it will save a weird filename with
f`{tempdir}/.part_{part_number}.{uuid.uuid4()}.tmp'. You may want to remove the `.` or add a prefix base on the actual filename?